bundles/org.openhab.binding.folderwatcher/src/main/resources/OH-INF/thing/thing-types.xml

```diff
@@ -0,0 +1,126 @@
+		<description>FTP folder to be watched</description>
+
+		<channels>
+			<channel id="newfile" typeId="newfile-channel"/>
+		</channels>
+
+		<config-description>
+			<parameter name="ftpAddress" type="text" required="true">
+				<label>FTP Server</label>
+				<description>Address of FTP server</description>
+				<context>network-address</context>
+			</parameter>
+			<parameter name="ftpPort" type="integer" min="1" max="65535">
+				<label>FTP Port</label>
+				<default>21</default>
+				<description>FTP server's port</description>
+			</parameter>
+			<parameter name="secureMode" type="text">
+				<label>FTP Security</label>
+				<limitToOptions>true</limitToOptions>
+				<options>
+					<option value="NONE">None</option>
+					<option value="IMPLICIT">TLS/SSL Implicit</option>
+					<option value="EXPLICIT">TLS/SSL Explicit</option>
+				</options>
+				<default>NONE</default>
+				<description>FTP Security settings</description>
+				<advanced>true</advanced>
+			</parameter>
+			<parameter name="ftpUsername" type="text" required="true">
+				<label>Username</label>
+				<description>User name</description>
+			</parameter>
+			<parameter name="ftpPassword" type="text" required="true">
+				<label>Password</label>
+				<description>FTP server password</description>
+				<context>password</context>
+			</parameter>
+			<parameter name="ftpDir" type="text" required="true">
+				<label>Root Directory</label>
+				<description>Root directory to be watched</description>
+			</parameter>
+			<parameter name="listHidden" type="boolean">
+				<label>List Hidden</label>
+				<default>false</default>
+				<description>Allow listing of hidden files</description>
+				<advanced>true</advanced>
+			</parameter>
+			<parameter name="listRecursiveFtp" type="boolean">
+				<label>List Sub Folders</label>
+				<default>false</default>
+				<description>Allow listing of sub folders</description>
+				<advanced>true</advanced>
+			</parameter>
+			<parameter name="connectionTimeout" type="integer" min="1" unit="s">
+				<label>Connection Timeout</label>
+				<description>Connection timeout for FTP request, sec</description>
+				<default>30</default>
+				<advanced>true</advanced>
+			</parameter>
+			<parameter name="pollInterval" type="integer" min="1" unit="s">
+				<label>Polling Interval</label>
+				<description>Interval for polling folder changes, sec</description>
+				<default>60</default>
+				<advanced>true</advanced>
+			</parameter>
+			<parameter name="diffHours" type="integer" min="1" unit="h">
+				<label>Timestamp Difference</label>
+				<description>How many hours back to analyze</description>
+				<default>24</default>
+				<advanced>true</advanced>
+			</parameter>
+		</config-description>
+
+	</thing-type>
+
+	<channel-type id="newfile-channel">
+		<kind>trigger</kind>
+		<label>New File Name(s)</label>
+		<description>A new file name</description>
+		<category>String</category>
+		<state readOnly="true"></state>
+	</channel-type>
```

Comment:
It seems to me rather an event Channel: https://www.openhab.org/docs/developer/bindings/thing-xml.html#trigger-channel-types

Reply:
changed